CMakeLists.txt

```diff
@@ -441,6 +441,18 @@ elseif(MSVC)
                             ${MSVC_LEVEL4_WARNINGS_LIST})
   set(CMAKE_C_FLAGS   "-utf-8 -Wall -WX ${MSVC_DISABLED_WARNINGS_STR} ${MSVC_LEVEL4_WARNINGS_STR}")
   set(CMAKE_CXX_FLAGS "-utf-8 -Wall -WX ${MSVC_DISABLED_WARNINGS_STR} ${MSVC_LEVEL4_WARNINGS_STR}")
+
+  # If we're using MSVC on Windows in FIPS mode with RelWithDebInfo then we want to replace the '-g' debug flag with '/Zi'.
+  # The '/Zi' flag accomplishes the same thing as '-g', but in a manner that is compatible with MSVC.
+  # This also fixes the problem we run into with RelWithDebInfo and FIPS mode on Windows where the FIPS module wouldn't span the expected symbol.
+  if(CMAKE_BUILD_TYPE_LOWER MATCHES "relwithdebinfo" AND FIPS)
+    message(STATUS "Replacing default -g flag for ${CMAKE_BUILD_TYPE} with /Zi")
+    string(REPLACE "-g" "/Zi" CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
```

Comment:
What is this doing? `-g` is a *nix build option to generate debug symbols. Right now on main a release with debug build uses these flags: `  FLAGS = -utf-8 -Wall -WX  -wd4061 -wd4100 -wd4127 -wd4200 -wd4204 -wd4221 -wd4242 -wd4244 -wd4267 -wd4371 -wd4388 -wd4296 -wd4350 -wd4365 -wd4389 -wd4464 -wd4510 -wd4512 -wd4514 -wd4548 -wd4610 -wd4623 -wd4625 -wd4626 -wd4628 -wd4668 -wd4706 -wd4710 -wd4711 -wd4800 -wd4820 -wd5026 -wd5027 -wd5045 -wd4255 -wd4152 -wd4295 -wd4701 -wd4505 -wd4702 -wd4191 -wd4996  -w44265 /MD /Zi /O2 /Ob1 /DNDEBUG` from build/build.ninja.

The link options are `  LINK_FLAGS = /machine:x64 /debug /INCREMENTAL`

Reply:
This was an oversight on my part. I didn't see `/Zi` anywhere in the cmake files so I thought it was still using `-g`. Turns out it works without this so will remove 😅